src/btouch.cs

```
@@ -376,16 +376,25 @@ static int Main2 (string [] args)
 			CurrentPlatform = PlatformName.MacOSX;
 			Unified = true;
 			nostdlib = true;
-			skipSystemDrawing = target_framework == TargetFramework.Xamarin_Mac_4_5_Full;
 			if (string.IsNullOrEmpty (baselibdll)) {
-				if (target_framework == TargetFramework.Xamarin_Mac_2_0_Mobile) {
+				if (target_framework == TargetFramework.Xamarin_Mac_2_0_Mobile)
 					baselibdll = Path.Combine (GetSDKRoot (), "lib", "reference", "mobile", "Xamarin.Mac.dll");
-					AddAnyMissingSystemReferencesFromSDK ("lib/mono/Xamarin.Mac", references);
-				} else {
+				else if (target_framework == TargetFramework.Xamarin_Mac_4_5_Full)
+					baselibdll = Path.Combine (GetSDKRoot (), "lib", "reference", "full", "Xamarin.Mac.dll");
+				else if (target_framework == TargetFramework.Xamarin_Mac_4_5_System)
 					baselibdll = Path.Combine (GetSDKRoot (), "lib", "reference", "full", "Xamarin.Mac.dll");
-					AddAnyMissingSystemReferencesFromSDK ("lib/mono/4.5", references);
-				}
 			}
+			if (target_framework == TargetFramework.Xamarin_Mac_2_0_Mobile) {
+				skipSystemDrawing = true;
+				AddAnyMissingSystemReferencesFromSDK ("lib/mono/Xamarin.Mac", references);
+			} else if (target_framework == TargetFramework.Xamarin_Mac_4_5_Full) {
+				skipSystemDrawing = true;
+				AddAnyMissingSystemReferencesFromSDK ("lib/mono/4.5", references);
+			} else if (target_framework == TargetFramework.Xamarin_Mac_4_5_System) {
+				skipSystemDrawing = false;
+				AddAnyMissingSystemReferences ("/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/4.5", references);
+			}
```

Same: `else`...?

If it's not possible, we should throw an exception.